src/main/java/seedu/address/MainApp.java

```diff
@@ -128,6 +128,10 @@ protected Config initConfig(Path configFilePath) {
             logger.warning("Config file at " + configFilePathUsed + " is not in the correct format."
                     + " Using default config properties.");
             initializedConfig = new Config();
+        } catch (IOException e) {
+            logger.warning("Problem while reading from the file " + configFilePathUsed
+                    + ". Will be starting with default configuration");
+            initializedConfig = new Config();
```

Comment:
Is it possible to combine these two catch clauses, as the action taken is the same?  

```java
        } catch (DataConversionException | IOException e) {
            logger.warning(e.getMessage + " Using default config properties.");
            initializedConfig = new Config();
        }
```

Reply:
`DataConversionException` is a custom exception that actually takes in a file path. So when `e.getMessage()` is called on it, it returns a file path instead of the exception as compared to `IOException`.

Reply:
> `DataConversionException` is a custom exception that actually takes in a file path.

I don't think so. Check the code.

Reply:
>so it might better if we just provide a short description of what happened instead?

How would the code look like for that option?

Reply:
DataConversionException: "XXX file at <file location> is might be corrupted/in the wrong format\n"
IOException: "An error occurred while reading from XXX file at <file location>: <e.getMessage()>\n"

We can perhaps combine them into a singular catch block like so:

```java
      }  catch (DataFormatException | IOException e) {
            String description = "";
            if (e instanceof DataFormatException) {
                description = "Config file at "+ configFilePathUsed +" is might be corrupted or in the wrong format\n";
            } else {
                description = "An error occurred while reading from XXX file at <file location>\n";
            }
            logger.warning(description + "Using default config properties.");
            initializedConfig = new Config();
        }
```

Reply:
`if (e instanceof DataConversionException)` defeats the purpose of combining the two `catch` clauses. In any case, I just noticed that the `initModelManager` method uses two `catch` clauses as well. So, we can do the same here.